agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/localstorage/LocalFileLoader.java

```diff
@@ -92,18 +100,53 @@ PersistedFile loadTelemetriesFromDisk() {
       return null;
     }
 
-    byte[] result;
-    try {
-      // TODO (trask) optimization: read this directly into ByteBuffer(s)
-      result = Files.readAllBytes(tempFile.toPath());
+    byte[] ikeyBytes = new byte[36];
+    int rawByteLength = (int) tempFile.length() - 36;
+    byte[] telemetryBytes = new byte[rawByteLength];
```

Comment:
```suggestion
    if (tempFile.length() <= 36) {
       ... just in case ...
       operationLogger.recordFailure(...);
       return null;
    }
    byte[] ikeyBytes = new byte[36];
    int rawByteLength = (int) tempFile.length() - 36;
    byte[] telemetryBytes = new byte[rawByteLength];
```

Reply:
it shouldn't happen.. but if it does somehow like corrupted data, it should be deleted then?

Reply:
👍